### ISSUE TITLE
The invoked member is not supported before the type is created.

### ISSUE THREAD

**Issue by albertwoo** [Microsoft/aspnet-api-versioning#431]
Step to reproduce:
1. In /src/samples/aspnetcore/SwaggerODataSample/Models/Person.cs, add property `public Bug Bug { get; set; }` to Person class. Bug class is like:
```c#
    public class Bug
    {
        public int Id { get; set; }
        public string Detail { get; set; }
    }
```
2. Run the project and I got Exception1.
3. This line https://github.com/Microsoft/aspnet-api-versioning/blob/509bdfd63c4a9a7fc54c6151a405f07bd9adceb3/src/Common.OData.ApiExplorer/AspNet.OData/StructuredTypeResolver.cs#L20 has a bug. we should change to something like:
```c#
var structuredType = structuredTypes.FirstOrDefault( t => type.FullName == t.GetClrType( model )?.FullName );
```
4. After we fix this, the project can run now, but in the swagger ui, if we switch to v3, we got Exception2

Exception1:
```
   at System.Reflection.Emit.TypeBuilder.GetCustomAttributes(Type attributeType, Boolean inherit)
   at System.Attribute.GetCustomAttributes(MemberInfo element, Type type, Boolean inherit)
   at System.Attribute.GetCustomAttribute(MemberInfo element, Type attributeType, Boolean inherit)
   at System.Reflection.CustomAttributeExtensions.GetCustomAttribute[T](MemberInfo element, Boolean inherit)
   at Microsoft.AspNet.OData.StructuredTypeResolver.GetStructuredType(Type type) in C:\Users\binwwu\Documents\Github\aspnet-api-versioning\src\Common.OData.ApiExplorer\AspNet.OData\StructuredTypeResolver.cs:line 28
   at Microsoft.AspNet.OData.Builder.ODataAttributeVisitor.Visit(ApiDescription apiDescription) in C:\Users\binwwu\Documents\Github\aspnet-api-versioning\src\Common.OData.ApiExplorer\AspNet.OData\Builder\ODataAttributeVisitor.cs:line 63
   at Microsoft.AspNet.OData.Builder.ODataValidationSettingsConvention.ApplyTo(ApiDescription apiDescription) in C:\Users\binwwu\Documents\Github\aspnet-api-versioning\src\Microsoft.AspNetCore.OData.Versioning.ApiExplorer\AspNet.OData\Builder\ODataValidationSettingsConvention.cs:line 38
   at Microsoft.AspNet.OData.Builder.ODataControllerQueryOptionConvention.ApplyTo(ApiDescription apiDescription) in C:\Users\binwwu\Documents\Github\aspnet-api-versioning\src\Common.OData.ApiExplorer\AspNet.OData\Builder\ODataControllerQueryOptionConvention.cs:line 45
   at Microsoft.AspNet.OData.Builder.ODataQueryOptionsConventionBuilder.ApplyTo(IEnumerable`1 apiDescriptions, ODataQueryOptionSettings queryOptionSettings) in C:\Users\binwwu\Documents\Github\aspnet-api-versioning\src\Common.OData.ApiExplorer\AspNet.OData\Builder\ODataQueryOptionsConventionBuilder.cs:line 156
   at Microsoft.AspNetCore.Mvc.ApiExplorer.ODataApiDescriptionProvider.ExploreQueryOptions(IEnumerable`1 apiDescriptions, ODataUriResolver uriResolver) in C:\Users\binwwu\Documents\Github\aspnet-api-versioning\src\Microsoft.AspNetCore.OData.Versioning.ApiExplorer\AspNetCore.Mvc.ApiExplorer\ODataApiDescriptionProvider.cs:line 278
   at Microsoft.AspNetCore.Mvc.ApiExplorer.ODataApiDescriptionProvider.OnProvidersExecuted(ApiDescriptionProviderContext context) in C:\Users\binwwu\Documents\Github\aspnet-api-versioning\src\Microsoft.AspNetCore.OData.Versioning.ApiExplorer\AspNetCore.Mvc.ApiExplorer\ODataApiDescriptionProvider.cs:line 207
   at Microsoft.AspNetCore.Mvc.ApiExplorer.ApiDescriptionGroupCollectionProvider.GetCollection(ActionDescriptorCollection actionDescriptors)
   at Microsoft.AspNetCore.Mvc.ApiExplorer.ApiDescriptionGroupCollectionProvider.get_ApiDescriptionGroups()
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GetSwagger(String documentName, String host, String basePath, String[] schemes)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
```

Exception2:
```
System.NotSupportedException: The invoked member is not supported before the type is created.
   at System.Reflection.Emit.TypeBuilder.GetCustomAttributes(Boolean inherit)
   at Newtonsoft.Json.Utilities.ReflectionUtils.GetAttributes(Object attributeProvider, Type attributeType, Boolean inherit)
   at Newtonsoft.Json.Serialization.JsonTypeReflector.GetAssociateMetadataTypeFromAttribute(Type type)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Newtonsoft.Json.Serialization.JsonTypeReflector.GetAttribute[T](Type type)
   at Newtonsoft.Json.Serialization.JsonTypeReflector.GetAttribute[T](Object provider)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Newtonsoft.Json.Serialization.DefaultContractResolver.CreateContract(Type objectType)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Newtonsoft.Json.Serialization.DefaultContractResolver.ResolveContract(Type type)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaRegistry.CreateSchema(Type type, Queue`1 referencedTypes)
   at Swashbuckle.AspNetCore.SwaggerGen.SchemaRegistry.GetOrRegister(Type type)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.CreateBodyParameter(ApiParameterDescription apiParameterDescription, String name, Boolean isRequired, ISchemaRegistry schemaRegistry)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.CreateParameter(ApiDescription apiDescription, ApiParameterDescription apiParameterDescription, ISchemaRegistry schemaRegistry)
   at System.Linq.Enumerable.WhereSelectListIterator`2.ToList()
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.CreateOperation(ApiDescription apiDescription, ISchemaRegistry schemaRegistry)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.CreatePathItem(IEnumerable`1 apiDescriptions, ISchemaRegistry schemaRegistry)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.CreatePathItems(IEnumerable`1 apiDescriptions, ISchemaRegistry schemaRegistry)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GetSwagger(String documentName, String host, String basePath, String[] schemes)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Server.IIS.Core.IISHttpContextOfT`1.ProcessRequestAsync()
```

**Comment by LuukN2** [comment]
This seems like same issue I had which should be fixed with the change in #425 

Since it appears to be an issue that is blocking users I'll bump up the patch version so it can be released as a patch.

If it is possible for you to build that branch and see if it fixes your issue that would be helpful.

**Comment by commonsensesoftware** [comment]
The fix for this issue has been published in [3.1.1](https://www.nuget.org/packages/Microsoft.AspNetCore.OData.Versioning.ApiExplorer/3.1.1). It might take a few minutes for indexing to complete. If the changes somehow do not address this issue, please reopen it. Thanks.
